18.3w fb watermonitoring

dnicolalde opened the pull request:
Merging water monitoring with SNAPSHOT. We can split this merge between Gabe and Chad.

dnicolalde committed:
Changes to water monitoring module from 17.2 version

dnicolalde committed:
Changes made based on recomendations from Trey, ticket# 36961

dnicolalde committed:
Revert "Changes made based on recomendations from Trey, ticket# 36961"

This reverts commit ff437ee9

dnicolalde committed:
Change to webUtils to allow parameters.

Commit:
Adding validation to check if animals are in a water restricted protocol. Animals need to be added to waterRestrictedAnimals table.

Commit:
Including FullCalendar 3.10 which is needed for the webPart

Commit:
Adding dependency for fullCalendar and fixing typos.

dnicolalde committed:
Move code from 17.2 missing from initial code merge.

dnicolalde committed:
Updating LabKey version to add species to supplemental pedigree.

dnicolalde committed:
Ticket 39768: Error with JavaScript preventing ext4 files to load.

dnicolalde committed:
Adding link to water calendar to ehrBegin page.

dnicolalde committed:
Changing labels for water entry and removing restricted to scheduled.

Commit:
Updating validation for water scheduled animals

Commit:
Changing water order form title and adding report to show water in grid format.

dnicolalde committed:
Changing Lab Water title for future water orders

Commit:
Grid view for future waters and working on the binding problem with knockout

Commit:
Changes to PdfInvoice

Commit:
Changes to PdfInvoice

Commit:
Changing label on volume

Commit:
Minor changes to display labels.

Commit:
removing extra button from tbar

Commit:
removing template from food deprives

Commit:
adding provide fruit to water amount, orders and given, and removing several columns. Duplicating if provide fruit is selected

Commit:
removing error on Arrays import

Commit:
Adding button to update time and arranging column order.

Commit:
Adding validation to not allow end time be before start time.

Commit:
Changes to EXT to hide remarks, changes to action from calendar view.

Commit:
Making small changes to water system

Commit:
Adding changes to Water order form.

Commit:
Solve problem with re-rendering webpart with knockout and full calendar

Commit:
Fixing problems with validation with overlapping waters orders. Fixed problem with production rendering of water calendar.

Commit:
Solve problem with seconds showing in chair restraint section of water monitoring.

Commit:
Fixing various problems with water monitoring system.

Commit:
Default value for water amounts

Commit:
Adding waterSource to waterSchedule

Commit:
Add Lixt and Regulated to water system

Commit:
Changes to water monitoring system, adding drug section and making sure the water given is only ones.

Commit:
Merge remote-tracking branch 'remotes/origin/wnprc18.3-SNAPSHOT' into 18.3w_fb_watermonitoring

# Conflicts:
#	WNPRC_EHR/src/org/labkey/wnprc_ehr/WNPRC_EHRController.java

Commit:
Adding drug section for implant cleaning.

Commit:
Fixing chairing validation to convert times correctly.

Commit:
Trying to solve problem with editing an existing water giving task

Commit:
solve problem with re-rendering webpart with

Commit:
Revert "Trying to solve problem with editing an existing water giving task"

This reverts commit 6c1b52e6

Commit:
Fixing problem with masterRecord on water form panel.

Commit:
Merge branch 'wnprc18.3-SNAPSHOT' into 18.3w_fb_watermonitoring

Commit:
Fixes to water monitoring system, adding water status.

Commit:
Improving validation for adding water orders when water mounts are in the system.

Commit:
Merge branch 'wnprc18.3-SNAPSHOT' into 18.3w_fb_watermonitoring

# Conflicts:
#	package-lock.json

Commit:
Adding return error messages to Bootstrap when changes water orders.

Commit:
Default time to request water order starting the next day.

Commit:
Updating water features based on feedback from Sept 25th

Commit:
Changes to WaterMonitoring, removing additional water from waterOrders

Commit:
Changes singleanimalEncounter to use clinical encounter.

Commit:
Merge remote-tracking branch 'remotes/origin/wnprc18.3-SNAPSHOT' into 18.3w_fb_watermonitoring

# Conflicts:
#	WNPRC_EHR/resources/queries/study/restraints.js
#	WNPRC_EHR/resources/queries/study/restraints.query.xml
#	WNPRC_EHR/resources/views/dataEntry.html
#	WNPRC_EHR/resources/views/ehrBegin.html
#	WNPRC_EHR/resources/web/wnprc_ehr/wnprc_ext4.lib.xml
#	WNPRC_EHR/src/org/labkey/wnprc_ehr/TriggerScriptHelper.java
#	WNPRC_EHR/src/org/labkey/wnprc_ehr/WNPRC_EHRModule.java
#	WNPRC_EHR/src/org/labkey/wnprc_ehr/table/WNPRC_EHRCustomizer.java

Commit:
Closing Lixit order when entering a new valid water order, also changing all water orders if switch to Lixit.

Commit:
Adding default value for water order assignedto to researchStaff

Commit:
Changes to allow water given to be submitted and also allowing to update existing orders that are in lixit status.

Commit:
Fixing problem with water given trigger script, started to move validation to waterAmount triggerScript.

Commit:
Adding provideFruit to carry over from waterOrder to waterAmount
Show On Lixit if volume is set to null

Commit:
Update fullCalendar to be included in JSP page and system now adds volumes from different sources (i.e. waterOrder and waterAmount)

Commit:
Fixing problems with fullCalendar, not reloading jQuery a second time and unbindding components if the housing and asignments report is display.

Commit:
Change how wnprcReports.js calls for dependencies in webUtils.

Commit:
Updating links acccording to Issue 9366

Commit:
Update to water calendar and water pivot to adhere to a variable mlsperKg

Commit:
Adding a view for active water orders, and fixing problem with showing entire db in water calendar

Commit:
Fixing problem with closing water orders and opening new order

dnicolalde committed:
Adding frequency and improving error handling in water monitoring

dnicolalde committed:
Removing logging output.

dnicolalde committed:
UpdateToFullCalendar5 and fixing display issues with bootstrap display

dnicolalde committed:
Fixing problem with JSP rendering of Javascript console log

dnicolalde committed:
Fixing problem with JSP rendering of Javascript console log

dnicolalde committed:
Adding change for userAccess.

dnicolalde committed:
Fixing to knockout bindings.

dnicolalde committed:
Adding fullCalendar 5.3.2

dnicolalde committed:
Adding water prePivot to fullHistory using a UNION ALL

dnicolalde committed:
Fixing name of water total for the day, keeping with filter on the .qview.xml

dnicolalde committed:
Improving access control to allow vet and compliance to see water records.

dnicolalde committed:
Reducing the drop down options for insert and single day water in the WaterCalendar view

dnicolalde committed:
Adding dirty flag to improve the UI

dnicolalde committed:
Resetting dirty flag after updating or adding a water amount.

dnicolalde committed:
Adding better message for animals not in water restriction and adding addition validation for Lab water

dnicolalde committed:
Correcting validation for animals not in the water Schedule animals

dnicolalde committed:
Fixing waterGiven events on the Calendar view.

dnicolalde committed:
Fixing problem with waterAmounts added from LabWater form, setting time to 14:01

dnicolalde committed:
Removing dirty flag text

dnicolalde committed:
New client store extending SlaveSectionClientStore.js

dnicolalde committed:
Changes to the stores to check date and time from encounter

dnicolalde committed:
Merge branch 'wnprc18.3-SNAPSHOT' into 18.3w_fb_watermonitoring

# Conflicts:
#	WNPRC_EHR/resources/views/dataEntry.html

dnicolalde committed:
Change on name for class to Apprentice

dnicolalde committed:
Adding notification for the water system and adding a new drug.js and query.xml for customized WNPRC behavior.

dnicolalde committed:
Improving notification for water monitoring system adding table with animalIds to email

dnicolalde committed:
Adding a selection for recurring events and fixing the on lixit

dnicolalde committed:
Commit checkOldVendorIds method added by Chad

dnicolalde committed:
Merge remote-tracking branch 'origin/wnprc18.3-SNAPSHOT' into 18.3w_fb_watermonitoring

# Conflicts:
#	WNPRC_EHR/resources/views/dataEntry.html
#	WNPRC_EHR/src/org/labkey/wnprc_ehr/TriggerScriptHelper.java
#	WNPRC_EHR/src/org/labkey/wnprc_ehr/table/WNPRC_EHRCustomizer.java

csebranek committed:
Comment out console logs in chairing trigger script

csebranek commented on WNPRC_EHR/resources/queries/study/drug.js:
```diff
@@ -0,0 +1,73 @@
+    helper.decodeExtraContextProperty('clientEncounterDate');
+
+    helper.registerRowProcessor(function(helper,row) {
+        if (!row)
```
@dnicolalde The code here I've seen in more than 2 trigger scripts, could probably be ripped out and placed into a reusable "utils" function but just going to add a TODO note here for now.

csebranek committed:
Comments in drug trigger script

csebranek commented on WNPRC_EHR/resources/queries/study/remainingWater/.qview.xml:
```diff
@@ -0,0 +1,17 @@
+   <columns>
+       <column name="Id" />
+       <column name="date" />
+       <column name="volumeGivenByLab" />
```
@dnicolalde do these columns match up to the query? In the remainingWater.sql file I noticed here it is called 'volumeGivenByLab' but in the query it is called 'volumeGivenInLab' ?

dnicolalde replied:
this an older code deleted from the branch.

gottfred reviewed (approved):
All of the fullcalendar related stuff looks reasonable. We'll have to strip out the fullcalendar libs when we merge into 21.3 (and update the jsp to reference the new npm sourced ones), but seems good for 18.3

csebranek committed:
Remove console logs from waterAmount trigger script

csebranek commented on WNPRC_EHR/resources/queries/study/waterAmount.js:
```diff
@@ -0,0 +1,125 @@
+        //console.log("value of fixdate " + row.date);
+        if (!row.skipWaterRegulationCheck){
+            //TODO: look up the qcstate given the label
+            row.qcstate = 10;
```
@dnicolalde Would probably be best to look up the qc state given the label, something like EHR.Security.getQCStateMap() might do the trick.

dnicolalde replied:
Change to what you are recommending.

dnicolalde replied:
server side we have to use: EHR.Server.Security.getQCStateByLabel('Scheduled').Label or
EHR.Server.Security.getQCStateByLabel('Scheduled').rowid

csebranek committed:
Cleaning up console logs in waterOrders trigger script

In general, cleaning up console logs that would be difficult to track down in production, leaving the ones that could provide useful information for debugging

csebranek commented on WNPRC_EHR/resources/queries/study/waterPivot.query.xml:
```diff
@@ -0,0 +1,60 @@
+
+
+                    </column>
+                    <column columnName="r12012::total">
```
@dnicolalde What does this do?

dnicolalde replied:
old code, deleted.

csebranek commented on WNPRC_EHR/resources/queries/study/waterSchedule.query.xml:
```diff
@@ -0,0 +1,192 @@
+                        <isHidden>true</isHidden>
+                        <isKeyField>true</isKeyField>
+                    </column>
+                    <column columnName="primaryKey2">
```
@dnicolalde is primaryKey2 used?

dnicolalde replied:
code that came from treatment schedule, no longer needed

csebranek commented on WNPRC_EHR/resources/queries/study/water_amount.query.xml:
```diff
@@ -0,0 +1,25 @@
+<query xmlns="http://labkey.org/data/xml/query">
+    <metadata>
+        <tables xmlns="http://labkey.org/data/xml">
+            <table tableName="Water Amount" tableDbType="TABLE">
```
@dnicolalde is this dataset different than waterAmount?

dnicolalde replied:
this an old version, I should remove it.

csebranek committed:
Remove debugger breakpoint from animalWaterCalendar

csebranek committed:
Remove debugger from water grid calendar

csebranek commented on WNPRC_EHR/resources/web/wnprc_ehr/wnprcReports.js:
```diff
@@ -1329,7 +1329,7 @@ EHR.reports['abstract'] = function(panel, tab){
 
         LABKEY.Utils.requiresCSS("wnprc_ehr/HousingAndAssignmentHistory.css");
-        WNPRC_EHR.Utils.Lib.loadLibrary(['/webutils/lib/webutils'], function() {
+        LABKEY.requiresScript('/webutils/lib/webutils', function() { 
```
@dnicolalde reason for the change? i assume this report functions the same, though.

dnicolalde replied:
Is a change on how webutils calls for dependencies.

csebranek committed:
Clean up imports in wnprc module

csebranek committed:
Remove incorrect authorship from water monitoring notification

csebranek committed:
Clean up imports in wnprc customizer

csebranek committed:
Clean up imports in wnprc customizer

csebranek commented on WNPRC_EHR/resources/queries/study/Water Request.js:
```diff
@@ -0,0 +1,50 @@
+        console.log("value of qcstate "+ row.QCStateLabel+ "  QCState Using Secuirty "+ EHR.Server.Security.getQCState(row)['isPublicData']);
+        if (row.QCStateLabel=='Completed'){
+            console.log ("one record completed "+ row.Id);
+            row.QCState = 1;
```
@dnicolalde is completed always qc state 1? EHR.Security.getQCStateMap is a better approach.

dnicolalde replied:
Old code deleted.

csebranek commented on WNPRC_EHR/src/org/labkey/wnprc_ehr/pages/husbandry/WaterCalendar.jsp:
```diff
@@ -0,0 +1,1540 @@
+        husbandryAssignmentLookup.put(map.get("value"), waterInfo);
+    }
+
+    Group vetGroup = GroupManager.getGroup(getContainer(), "veterinarians (LDAP)", GroupEnumType.SITE);
```
Might be good to use labkey roles here instead of hard coding groups @dnicolalde

dnicolalde replied:
We should add this as a TODO

csebranek commented on WebUtils/resources/web/webutils/lib/fullcalendar-3.10.0/CHANGELOG.txt:
```diff
@@ -0,0 +1,1342 @@
+
```
@dnicolalde is fullcalendar 3 being used?

csebranek committed:
Update logging on water request.js

csebranek committed:
Remove duplicate imports in wnprc controller

dnicolalde committed:
Changes based on review by Chad and Gabe

dnicolalde committed:
Merge branch '18.3w_fb_watermonitoring' of https://github.com/WNPRC-EHR-Services/wnprc-modules into 18.3w_fb_watermonitoring

 Conflicts:
	WNPRC_EHR/resources/queries/study/Water Request.js
	WNPRC_EHR/resources/queries/study/waterAmount.js

dnicolalde commented:
Thank you for checking my code.
